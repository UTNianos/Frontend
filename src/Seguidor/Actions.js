import { put } from 'redux-saga/effects';

import estadosData from './data/estados.json';
import materiasData from './data/materias.json';
import correlativasData from './data/correlativas.json';

export const REQUEST_MATERIAS = 'REQUEST_MATERIAS';
export const RECEIVE_MATERIAS = 'RECEIVE_MATERIAS';
export const RECEIVE_MATERIAS_FAILURE = 'RECEIVE_MATERIAS_FAILURE';

export const REQUEST_ESTADO = 'REQUEST_ESTADO';
export const RECEIVE_ESTADO = 'RECEIVE_ESTADO';
export const REQUEST_ESTADO_FAILURE = 'REQUEST_ESTADO_FAILURE';

export const REQUEST_CORRELATIVAS = 'REQUEST_CORRELATIVAS';
export const RECEIVE_CORRELATIVAS = 'RECEIVE_CORRELATIVAS';
export const RECEIVE_CORRELATIVAS_FAILURE = 'RECEIVE_CORRELATIVAS_FAILURE';

export const UPDATE_ESTADO = 'UPDATE_ESTADO';
export const UPDATE_ESTADO_OK = 'UPDATE_ESTADO_OK';
export const UPDATE_ESTADO_FAILURE = 'UPDATE_ESTADO_FAILURE';

export function* loadEstado(action) {
  try{
    yield put({type: RECEIVE_ESTADO, data: estadosData});
  }
  catch(error) {
    yield put({type: REQUEST_ESTADO_FAILURE })
  }
}

export function* loadMaterias(action) {
  try{
    yield put({type: RECEIVE_MATERIAS, data: materiasData});
  }
  catch(error) {
    yield put({type: RECEIVE_MATERIAS_FAILURE })
  }
}

export function* loadCorrelativas(action) {
  try{
    yield put({type: RECEIVE_CORRELATIVAS, data: correlativasData});
  }
  catch(error) {
    yield put({type: RECEIVE_CORRELATIVAS_FAILURE })
  }
}

export function* updateEstadoMateria(action) {
  try{
    const newStatus = parseInt(action.materiaEstado.status, 10);
    const newEstado = { id: action.materiaEstado.id, status: newStatus };

    yield put({type: UPDATE_ESTADO_OK, data: newEstado});
  }
  catch(error) {
    yield put({type: UPDATE_ESTADO_FAILURE, error: error})
  }
}
