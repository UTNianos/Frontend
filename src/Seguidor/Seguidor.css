.SeguidorHeading
{
  text-align: center;
  font-weight: 100;
  margin-bottom: 38px;
}

.ViewChooser
{
  margin-left: 77%;
  margin-bottom: 50px;
}

@media (min-width: 900px)
{
    .YearsVisualizer
	{
      display: flex;
	  width: 1145px;
	  margin: 0 auto;
	}
}