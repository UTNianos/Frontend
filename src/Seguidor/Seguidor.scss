.SeguidorHeading
{
  text-align: center;
  font-weight: 100;
  margin-bottom: 38px;
}
	

@media (min-width: 900px) 
{	
    .YearsVisualizer 
	{
	  display: flex;
	}
}