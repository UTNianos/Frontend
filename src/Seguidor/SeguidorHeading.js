import React from 'react';
import cssModules from 'react-css-modules';
import Styles from './Seguidor.scss';
import { Icon } from 'react-fa';
import { Button } from 'antd';

const SeguidorHeading = ({changeViewType, currentView}) => (
  <div>

    <div className="SeguidorTitle">
      <h1 styleName="SeguidorHeading">Materias</h1>
    </div>

    <div styleName="ViewChooser">
      <Button onClick={() => changeViewType('carousel')} >
          <Icon name="arrows-h" size="lg"/>
       </Button>
       <Button onClick={() => changeViewType('tree')}>
          <Icon name="tree" size="lg" />
       </Button>
       <Button onClick={() => changeViewType('finales')}>
          <Icon name="clock-o" size="lg" />
       </Button>
    </div>

  </div>
  )

export default cssModules(SeguidorHeading, Styles, { allowMultiple: true });
