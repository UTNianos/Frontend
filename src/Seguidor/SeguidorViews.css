.ViewChooser
{
  margin-left: 80%;
}

.IconStandard, .IconTree
{
  margin-right: 5px;
}
