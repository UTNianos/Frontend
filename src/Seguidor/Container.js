import { connect } from 'react-redux';
import Seguidor from './Seguidor';
import {
  loadEstado,
  loadMaterias,
  loadCorrelativas,
  updateEstadoMateria
} from './Actions';
import getMateriasEstado from './Selector';

const mapStateToProps = (state) => {
  return {
    materias: getMateriasEstado(state),
    isFetching: state.seguidor.isFetching,
    error: state.seguidor.error,
    estados: state.seguidor.estados
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    onLoad: () => {
      dispatch(loadEstado());
      dispatch(loadCorrelativas());
      dispatch(loadMaterias());
    },
    updateEstado: (materiaEstado) => {
      dispatch(updateEstadoMateria(materiaEstado));
    }
  }
}

const SeguidorContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(Seguidor);

export default SeguidorContainer;
