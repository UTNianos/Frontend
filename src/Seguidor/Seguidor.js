import React, { useState, useEffect } from 'react'
import MediaQuery from 'react-responsive';
import FetchingIndicator from '../Fetching/FetchingIndicator'
import SeguidorHeading from './SeguidorHeading'
import SeguidorView from './SeguidorView';

const Seguidor = (props) => {

  const {
    isFetching,
    error,
    materias,
    updateEstado
  } = props;

  useEffect(() => {
    if (materias.length === 0) { props.onLoad(); }

  }, [props, materias])

  const [view, setView] = useState('carousel');

  const changeViewType = (type) => {
    setView(type);
  }

  if (isFetching) { return <FetchingIndicator />; }

  if (error) { return <p>Hubo un error recuperando las materias.</p>; }

  return (
    <div className="Seguidor">
      <div>
        <SeguidorHeading
          changeViewType={changeViewType}
          currentView={view}
        />
      </div>
      <SeguidorView
        materias={materias}
        updateEstado={updateEstado}
        view={view}
      />
    </div>
  );
}

export default Seguidor;
