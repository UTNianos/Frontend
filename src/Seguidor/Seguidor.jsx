import React, { Component, PropTypes } from 'react';
import { Link } from 'react-router';
import FetchingIndicator from '../Fetching/FetchingIndicator';
import SeguidorHeading from './SeguidorHeading';
import YearTabs from './YearTabs';

class Seguidor extends Component {

 constructor(props) {
    super(props)
 }

 componentDidMount() {
   this.props.onLoad();
 }

 render() {

  const { isFetching, error, materias, estados} = this.props;

  if(isFetching)
    return <FetchingIndicator />;

  if(error)
    return <p>Hubo un error recuperando las materias</p>;

  return(
  <div>

    <div>
      <SeguidorHeading />
    </div>

    <div>
      <YearTabs materias={materias} updateFn={this.props.updateEstado} />
    </div>

  </div>
  )
 }
}

export default Seguidor;
