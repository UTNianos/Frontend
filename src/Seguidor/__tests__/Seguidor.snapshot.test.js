import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store' // ES6 modules
import { Provider } from 'react-redux';
import Seguidor from '../Seguidor';

const middlewares = []
const mockStore = configureStore(middlewares)

jest.mock('react-dom');
jest.mock('antd');

describe('<Seguidor />', () => {

  it('Renders correctly', () => {

    const materias = [{ cursada: false, final: false, id: '416', name: 'Materia 1', status: 1, year: '0' }, { cursada: false, final: false, id: '415', name: 'Materia 2', status: 1, year: '0' }];

    const correlativas = [{
      m: 1,
      d: {
        rCC: ['1', '2'],
        rFF: ['1', '2'],
        hCC: ['15', '71', '17', '72', '75', '20', '21', '22', '23', '24', '25', '26', '33'],
        hFF: ['15', '71', '17', '72', '75', '20', '21', '22', '23', '24', '25', '26', '33'],
        hFC: ['27', '73', '446', '447']
      }
    }];

    const estados = [{
      id: 416,
      status: 2
    },
    {
      id: 415,
      status: 3
    }
    ];

    const seguidorState = {
      materias,
      correlativas,
      estados,
      isFetching: false,
      error: false
    };

    const onLoadFn = jest.fn();
    const updateEstadoFn = jest.fn();

    const initialState = { seguidor: seguidorState };
    const store = mockStore(initialState);

    const SeguidorElement = () => (
      <Provider store={store} >
        <Seguidor
          onLoad={onLoadFn}
          updateEstado={updateEstadoFn}
          materias={materias}
          estados={estados}
          correlativas={correlativas}
        />
      </Provider>
    );

    const tree = renderer.create(<SeguidorElement />).toJSON();

    expect(tree).toMatchSnapshot();

  })

});
