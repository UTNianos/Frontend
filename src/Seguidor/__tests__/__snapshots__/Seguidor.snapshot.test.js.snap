// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Seguidor /> Renders correctly 1`] = `
<div>
  <div>
    <div>
      <div
        className="SeguidorTitle"
      >
        <h1
          className="SeguidorHeading"
        >
          Materias
        </h1>
      </div>
      <div
        className="ViewChooser"
      />
    </div>
  </div>
  <div />
</div>
`;
